lib/pbench/test/unit/server/query_apis/test_datasets_update.py

```diff
@@ -228,10 +186,147 @@ def fake_bulk(
         monkeypatch.setattr("elasticsearch.helpers.streaming_bulk", fake_bulk)
 
         response = client.post(
-            f"{server_config.rest_uri}/datasets/publish/random_md5_string1",
+            f"{server_config.rest_uri}/datasets/random_md5_string1",
             headers={"authorization": f"Bearer {pbench_token}"},
             query_string=self.PAYLOAD,
         )
 
         # Verify the failure
         capinternal("Unexpected backend error", response)
+
+    @pytest.mark.parametrize(
+        "owner,access",
+        [
+            ("drb", "public"),
+            ("drb", "private"),
+            ("test", "public"),
+            ("test", "private"),
+            (None, "private"),
+            (None, "public"),
+            ("drb", None),
+        ],
+    )
+    def test_query_owner_publish(
```

Comment:
I think the previous incarnation of this test did something slightly different with the `owner` field -- previously it was used to determine the owner of the dataset which was operated on, so that we worked through the combinations of _requestor_ and _dataset owner_ (to try publishing things that we did or did not own).  I don't think the current test is still covering that, since it now always uses a dataset which is owned by `"drb"` (assuming that the test doesn't actually change it...), so in the valid, non-Admin case, the requestor always matches the current owner.

And, not to be pedantic or anything, but I think the two dimensions of the matrix parameterized by the `"owner"` and `"access"` values defined in this snippet should produce nine cases to test (either here or among the other tests):

|Case|`requested "owner"`|`requested "access"`|Notes|
|-|----------|-----------|----------------|
|1|`"drb"`|`"public"`|Change owner from "drb" to "drb" and mark "public"|
|2|`"drb"`|`"private"`|Change owner from "drb" to "drb" and mark "private"|
|3|`"drb"`|`None`|Change owner from "drb" to "drb" and leave access unchanged|
|4|`"test"`|`"public"`|Change owner from "drb" to "test" and mark "public"|
|5|`"test"`|`"private"`|Change owner from "drb" to "test" and mark "private"|
|6|`"test"`|`None`|Change owner from "drb" to "test" and leave access unchanged|
|7|`None`|`"public"`|Leave owner unchanged and mark "public"|
|8|`None`|`"private"`|Leave owner unchanged and mark "private"|
|9|`None`|`None`|`BAD REQUEST`|

We currently are testing case 9 in `test_no_query_params()`...would it be better to roll that case into this test?  We're not testing case 6...do we think it's effectively identical to case 3?

Regardless, I think we need to have another column in the matrix:
- The requestor (provided by the `build_auth_header` fixture):
  - `pbench_token`/`"drb"`
  - `pbench_admin_token`/Admin
  - `pbench_token_invalid`
  - None
- The owner of the dataset (provided by a new parameterization):
  - `"drb"`
  - `"test"`
- The proposed new owner (provided by the existing parameterization):
  - `"drb"`
  - `"test"`
  - None
- The proposed new access (provided by the existing parameterization):
  - `"public"`
  - `"private"`
  - None

Reply:
Note that the `build_auth_header` quadruples the iterations of each parametrization here by testing with `drb` authentication, with `admin` authentication, with invalid authentication, and with no authentication. Since the parametrization here is testing datasets owned by `drb` and `test`, it *is* testing the "no, you can't do that" categories for both `access` (not owner/admin) and `owner` (not admin), and the expected statuses reflect that.

Reply:
> Since the parametrization here is testing datasets owned by `drb` and `test`

But that's the thing, Dave, I don't think it _is_ doing that (any more), see [this code line](https://github.com/distributed-system-analysis/pbench/blob/61f1713ffe626b15ff00629141458fffcb90a853/lib/pbench/test/unit/server/query_apis/test_datasets_update.py#L250).

The corresponding line in the old test referenced the `owner` parameter; now that parameter is used for the _proposed_ owner, instead.

Reply:
>I don't think it's testing what happens when someone tries to change the access to a dataset that they don't own.

Yeah @webbnh. You were right I missed this test case . 

>The corresponding line in the old test referenced the owner parameter; now that parameter is used for the proposed owner, instead.

After adding  `"None"` parameterization for the `owner`, So  I hard coded `drb` datasets (I thought in `build_auth_header` it'll do user -`test`  authentication for `VALID` type)